src/GitHubTools/Functions/Public/Update-AzurePipelineDependencies.ps1

```diff
@@ -0,0 +1,93 @@
+Update-AzurePipelineDependencies -GitHubOrganisation MyOrganisation -RepositoryPrefix foo-
+
+.NOTES
+powershell-yaml powershell module is required. Install-Module -name powershell-yaml
+The powershell-yaml module and the .NET library the module wraps around is no longer being maintained https://github.com/aaubry/YamlDotNet/discussions/689
+
+#>
+function Update-AzurePipelineDependencies {
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSReviewUnusedParameter", "RepositoryPrefix", Justification = "False positive as rule does not know that Where-Object operates within the same scope")]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSUseDeclaredVarsMoreThanAssignments", "UpdatePipelineYml", Justification="False positive as rule does not know that variable is updated in ForEach-Object")]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSUseShouldProcessForStateChangingFunctions", "")]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSUseSingularNouns", "")]
+    param(
+        [Parameter(Mandatory = $false)]
+        [String]$GitHubOrganisation = "SkillsFundingAgency",
+        [Parameter(Mandatory = $false)]
+        [String]$RepositoryPrefix = "das-"
+    )
+
+    $Repos = Get-GitHubRepo -GitHubOrg $GitHubOrganisation | Where-Object { $_.name -match "^$RepositoryPrefix.+" -and $_.isArchived -eq $False }
+
+    $Repos | ForEach-Object {
+        $Repo = $_
+
+        Write-Host "Scanning $($Repo.name)"
+
+        $PipelineYml = Get-GitHubRepoFileContent -GitHubOrganisation $GitHubOrganisation -RepositoryName $Repo.name -FilePath "azure-pipelines.yml"
+
+        if (!$PipelineYml) { return }
+
+        $Pipeline = ConvertFrom-Yaml -Yaml $PipelineYml.Content
+
+        $UpdatePipelineYml = $false
+        $Pipeline.resources.repositories | Where-Object { $_.type -eq "github" } | ForEach-Object {
+            if ($_.ref -like "refs/tags/*") {
+                $TagName = $_.ref.replace("refs/tags/", "")
+                $DependencyOrg = $_.name.Split("/")[0]
+                $DependencyRepo = $_.name.Split("/")[1]
+                $Releases = Get-GitHubRepoRelease -GitHubOrganisation $DependencyOrg -RepositoryName $DependencyRepo
+                $OutOfDate = $Releases[0].tag_name -ne $TagName
+
+                if ($OutOfDate) {
+                    Write-Warning "$($_.name) dependency: $TagName should be $($Releases[0].tag_name): updating"
+                    $PipelineYml.Content = $PipelineYml.Content.Replace($TagName, $Releases[0].tag_name)
+                    $UpdatePipelineYml = $true
+                }
+                else {
+                    Write-Host "$($_.name) dependency up to date" -ForegroundColor Green
+                }
+            }
+            else {
+                Write-Warning "No tag reference for: $($_.name) - $($_.ref)"
+                return
+            }
+        }
+
+        if ($UpdatePipelineYml || $UpdateCodeBuildYml) {
```

Comment:
`$UpdateCodeBuildYml` doesn't exist in this script, this is a good reason not to suppress `PSUseDeclaredVarsMoreThanAssignments`